src/pages/spawn/spawnHost/spawnHostModal/PublicKeyForm.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useState } from "react";
+import styled from "@emotion/styled";
+import { Subtitle } from "@leafygreen-ui/typography";
+import Checkbox from "@leafygreen-ui/checkbox";
+import TextInput from "@leafygreen-ui/text-input";
+import { RadioBox } from "@leafygreen-ui/radio-box-group";
+import { Select, Input } from "antd";
+import { InputLabel } from "components/styles";
+import { PublicKey, PublicKeyInput } from "gql/generated/types";
+
+const { Option } = Select;
+const { TextArea } = Input;
+
+export type publicKeyStateType = {
+  publicKey: PublicKeyInput;
+  savePublicKey: boolean;
+};
+
+interface PublicKeyFormProps {
+  publicKeys: PublicKey[];
+  onChange: React.Dispatch<React.SetStateAction<publicKeyStateType>>;
+  data: publicKeyStateType;
+}
+export const PublicKeyForm: React.FC<PublicKeyFormProps> = ({
+  publicKeys,
+  onChange,
+  data,
+}) => {
+  const [addNewKey, setAddNewKey] = useState(false);
```

Comment:
this sounds like a function name. booleans should start with `is` or `has` or `should`